recipe/meta.yaml

```diff
@@ -1,45 +1,151 @@
-{% set name = "<PLEASE ADD PKG NAME>" %}
-{% set version = "<PLEASE ADD PKG VERSION>" %}
+{% set name = "recbole" %}
+{% set version = "1.2.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: <sha256>
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/recbole-{{ version }}.tar.gz
+    sha256: 132a8a706462b26bed82d409b67fbc09e90b3a49b607c7ba76c2f1718d8ddc17
+    patches:
+      - patches/0001-update-colorlog-pinning.patch
+      - patches/0002-update-randomstate-to-default_rng-for-hyperopt.patch
+  # use github release archive for tests
+  - url: https://github.com/RUCAIBox/RecBole/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: 1bc77a567083884b41ddd652b9b35078a0c3554c39ca38ab825ab2681eef138e
+    folder: gh_src
 
 build:
   number: 0
+  # s390x: kmeans-pytorch (numba) unavailable
+  # py>=312: ray unavailable
```

Comment:
Do you want to build ray so you can get 3.12? I suppose this depends on whether snowflake wants to pay for that.

Reply:
@danpetry  we are on hold for this one (see ticket). We do have ray in our pipeline, if it's out before we get the OK for this one then I will remove the skip

Reply:
`ray-tune` does not have yet `py==3.12` variant.

```
ray-tune                       2.6.3 py310hca03da5_2  pkgs/main
ray-tune                       2.6.3 py311hca03da5_2  pkgs/main
ray-tune                       2.6.3  py38hca03da5_2  pkgs/main
ray-tune                       2.6.3  py39hca03da5_2  pkgs/main
```

skipping `py==312` because this ticket was on hold waiting for feedback for too long already.